editor/components/post-publish-panel/toggle.js

```diff
@@ -42,6 +42,21 @@ function PostPublishPanelToggle( {
 		return <PostPublishButton forceIsDirty={ forceIsDirty } forceIsSaving={ forceIsSaving } />;
 	}
 
+	if ( isContributor ) {
```

Comment:
Now that we have some conditional logic in this file, would it be worthwhile to add a couple of unit tests?

Check out `post-publish-button` for the pattern to follow.

Reply:
Absolutely! I believe so.

I'm still working on it, taking some time as I never added unit tests previously. Looking into the pattern of `post-publish-button` for reference.

Thank you @danielbachhuber.

Reply:
Thank you @danielbachhuber !

Just pushed [a new commit](https://github.com/WordPress/gutenberg/pull/6729/commits/414897c7443eb0797e9daa50423cd9d71656a867) with the unit tests. I'm quite sure it's all wrong, looking forward to getting your help in it if possible. Thank you so much.

Reply:
Looks great! I added a few more tests in 625fcd9d5b2868d6493582371c14342d2b1d643d

Reply:
Awesome one 😍, thank you so much! ❤